src/components/shareable-url-modal/shareable-url-modal.js

```diff
@@ -125,83 +128,157 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
 
   const handleModalClose = () => {
     onToggleModal(false);
-    setDeploymentState('default');
+    if (deploymentState !== 'incompatible') {
+      setDeploymentState('default');
+    }
     setResponseError(null);
     setIsLoading(false);
     setResponseUrl(null);
     setIsLinkSettingsClick(false);
     setInputValues({});
     setIsFormDirty({
       bucket_name: false,
-      region: false,
+      platform: false,
+      endpoint: false,
     }); /* eslint-disable camelcase */
   };
 
+  /**
+   * Returns the modal heading based on the given type and deployment state.
+   * @param {string} type - The type of the modal heading.
+   * @returns {string|null} The modal heading text or null if deployment state is 'default'.
+   */
+  const getModalHeading = (type) => {
```

Comment:
[Nit] `getModalHeading` is a bit confusing to me. Can we rename to `getDeploymentStateTitle` (not sure of this either)